06_safe_remote_purchase/contract.etch

```diff
@@ -0,0 +1,123 @@
+//------------------------------------------------------------------------------
+//
+//   Copyright 2019 Fetch.AI Limited
+//
+//   Licensed under the Apache License, Version 2.0 (the "License");
+//   you may not use this file except in compliance with the License.
+//   You may obtain a copy of the License at
+//
+//       http://www.apache.org/licenses/LICENSE-2.0
+//
+//   Unless required by applicable law or agreed to in writing, software
+//   distributed under the License is distributed on an "AS IS" BASIS,
+//   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//   See the License for the specific language governing permissions and
+//   limitations under the License.
+//
+//------------------------------------------------------------------------------
+
+// This contract require release 0.4.0 or above
+// Note that this is experimental functionality
+function main()
+  // TBD
+endfunction
+
+// This adapts SafeRemotePurchase.vy at https://vyper.readthedocs.io/en/v0.1.0-beta.10/vyper-by-example.html#safe-remote-purchases to Etch.
+// Rundown of the transaction:
+// 1. Seller posts item for sale and posts safety deposit of double the item value.
+//    Balance is 2*value.
+//    (1.1. Seller can reclaim deposit and close the sale as long as nothing was purchased.)
+// 2. Buyer purchases item (value) plus posts an additional safety deposit (Item value).
+//    Balance is 4*value.
+// 3. Seller ships item.
+// 4. Buyer confirms receiving the item. Buyer's deposit (value) is returned.
+//    Seller's deposit (2*value) + items value is returned. Balance is 0.
+@payable  // NOT POSSIBLE YET
```

Comment:
"// NOT POSSIBLE YET" marks the features I would need to make this contract work.